src/main/java/seedu/duke/component/gates/Gate.java

```diff
@@ -0,0 +1,13 @@
+package seedu.duke.component.gates;
+
+public abstract class Gate {
+    protected int output;
+    protected int input;
+
+    public abstract int getOutput();
+
+    public void setInput(byte input) {
```

Comment:
Any reason why the `@param input` here is of `byte` type when everything else is `int`? Including that of `TwoInputGate.setSecondInput()`.

Reply:
Oops I had initially made it byte then changed everything else to int. Seemed to have missed this one out. Thanks for pointing out!